tests/test_core.py

```diff
@@ -783,6 +783,13 @@ def __init__(self):
         imageio.imwrite("foo.bmp", "asd")
 
 
+def test_imwrite_symbol_name():
+    name = '''#!~@$%^&`-+{};',.() []_=.jpg'''
```

Comment:
```suggestion
    if sys.platform == "linux":
        name = '''#!~:@$%^&`-+{};',.()? []_=.jpg'''
    elif sys.platform == "win32":
        name = '''#!~:@$%^&`-+{};',.() []_=.jpg'''
    elif sys.platform == "darwin":
        name = '''#!~@$%^&`-+{};',.()? []_=.jpg'''
    else:
        pytest.skip("Unknown OS.")
```

On linux `?` may be in the path, but is reserved for URIs. I think it makes sense to test this explicitly, too.

Same for `:` on any OS except MacOS.

Reply:
I think the symbol ':' should not be added to the win32 test, because ':' cannot appear arbitrarily in win32. It will cause the win32 test to fail.

Reply:
@One-sixth Good catch. I updated the suggestion.